src/hotspot/share/runtime/stubRoutines.cpp

```diff
@@ -307,6 +312,28 @@ void compiler_stubs_init(bool in_compiler_thread) {
   }
 }
 
+// Check for Flush-To-Zero mode
+
+// On some processors faster execution can be achieved by returning
+// zero for extremely small results, rather than an IEEE-754 denormal
+// number. This mode is not compatible with the Java Language
+// Standard.
+bool StubRoutines::FTZ_mode_enabled() {
+  // Quickly test to make sure denormals are correctly handled.
+
+  // _small_denormal is the smallest denormal number that has two bits
+  // set. _large_denormal is a number such that, when _small_denormal
+  // is added to it, must be rounded according to the mode. These two
+  // tests detect the rounding mode in use. If denormals are turned
+  // off (i.e. denormals-are-zero) FTZ mode is in use.
+
+  // We need the addition of _large_denormal and _small_denormal to be
+  // performed at runtime. Making _small_denormal volatile ensures
+  // that the following expression isn't evaluated at compile time:
+  return (_large_denormal + _small_denormal == _large_denormal
```

Comment:
As a possible future expansion, if there are cases where foreign or native code sets the rounding mode to something other than round to nearest, expressions in the same vein can be used to detect that case and restore that other aspect of the control word.